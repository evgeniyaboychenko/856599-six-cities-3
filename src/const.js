const CardType = {
  NEAR: `near-places`,
  CITY: `cities`,
  FAVORITES: `favorites`
};

const SortType = {
  DEFAULT: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`
};

const AppRoute = {
  SIGN_IN: `/login`,
  FAVORITES: `/favorites`,
  ROOM: `/offer/:id`,
  MAIN: `/`,
};

const ResponseErrorCode = {
  UNAUTHORIZED: 401,
  INVALID_REQUEST: 400,
  NOT_FOUND: 404,
  SERVER_IS_NOT_AVAILABLE: 500
};

export {CardType, SortType, AppRoute, ResponseErrorCode};
