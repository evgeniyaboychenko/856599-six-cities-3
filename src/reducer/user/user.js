import {extend} from '../../utils/utils.js';
import {adaptUser} from '../../utils/adapter.js';
import {apiDefault} from '../../api.js';
const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  error: ``,
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  user: {
    id: -1,
    name: ``,
    avatar: ``,
    email: ``,
    isPro: false,
  }
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_LOADING_ERROR: `SET_LOADING_ERROR`
};

const ActionCreator = {
  requireAuthorization: (status, user) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: {status, user}
    };
  },
  setLoadingError: (error) => {
    return {
      type: ActionType.SET_LOADING_ERROR,
      payload: error
    };
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload.status,
        user: action.payload.user
      });
    case ActionType.SET_LOADING_ERROR:
      return extend(state, {
        error: action.payload
      });
  }

  return state;
};

const Operation = {
  checkAuth: () => (dispatch) => {
    return apiDefault.get(`/login`)
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH, adaptUser(response.data)));
      }
      // .catch((err) => {
      //   throw err;
      // });
      , () => dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.NO_AUTH, {id: -1,
        name: ``,
        avatar: ``,
        email: ``,
        isPro: false})));
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH, adaptUser(response.data)));
      })
    .catch((err) => {
      const {response} = err;
      dispatch(ActionCreator.setLoadingError(response.data.error));
      // throw err;
    });
  },
};

export {reducer, ActionCreator, ActionType, AuthorizationStatus, Operation};
