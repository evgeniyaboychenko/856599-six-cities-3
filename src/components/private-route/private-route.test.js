import React from 'react';
import renderer from 'react-test-renderer';
import {PrivateRoute} from './private-route.jsx';
import {BrowserRouter} from 'react-router-dom';
import {AuthorizationStatus} from '../../reducer/user/user.js';
import {AppRoute} from '../../const.js';

it(`PrivateRoute component render`, () => {
  const tree = renderer
  .create(
      <BrowserRouter>
        <PrivateRoute
          render = {() => {}}
          path = {AppRoute.FAVORITES}
          exact = {true}
          authorizationStatus = {AuthorizationStatus.AUTH}
        />
      </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
