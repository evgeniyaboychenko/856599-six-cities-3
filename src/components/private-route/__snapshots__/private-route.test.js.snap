// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute component render 1`] = `null`;
